omx/gstomxaudioenc.c

```diff
@@ -644,23 +633,48 @@ gst_omx_audio_enc_set_format (GstAudioEncoder * encoder, GstAudioInfo * info)
   GstOMXAudioEnc *self;
   GstOMXAudioEncClass *klass;
   gboolean needs_disable = FALSE;
-  OMX_PARAM_PORTDEFINITIONTYPE port_def;
-  OMX_AUDIO_PARAM_PCMMODETYPE pcm_param;
   gint i;
+  guint n_samples;
+  OMX_PARAM_PORTDEFINITIONTYPE port_def_in;
+  OMX_PARAM_PORTDEFINITIONTYPE port_def_out;
+  OMX_AUDIO_PARAM_PCMMODETYPE pcm_param;
+  OMX_PORT_PARAM_TYPE param;
   OMX_ERRORTYPE err;
 
   self = GST_OMX_AUDIO_ENC (encoder);
   klass = GST_OMX_AUDIO_ENC_GET_CLASS (encoder);
 
   GST_DEBUG_OBJECT (self, "Setting new caps");
 
+  n_samples =
```

Comment:
We should check if the `klass` has a `get_num_samples` implementation. Otherwise this is a potential seg fault. Even if there is a default installed, someone could've set it to `NULL` for example.